components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/handler/RegistrationHandlerTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * you may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.dcr.handler;
+
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.carbon.identity.oauth.dcr.context.DCRMessageContext;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationRequest;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationRequestProfile;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationResponse;
+import org.wso2.carbon.identity.oauth.dcr.model.RegistrationResponseProfile;
+import org.wso2.carbon.identity.oauth.dcr.service.DCRManagementService;
+import org.wso2.carbon.identity.testutil.powermock.PowerMockIdentityBaseTest;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+
+@PrepareForTest({DCRManagementService.class, RegistrationHandler.class})
+public class RegistrationHandlerTest extends PowerMockIdentityBaseTest {
+
+    private RegistrationHandler registrationHandler;
+    private String testTenantDomain = "testTenantDomain";
+
+    @Mock
+    RegistrationRequest mockRegisterRequest;
+
+    @Mock
+    DCRMessageContext mockDcrMessageContext;
+
+    @Mock
+    RegistrationRequestProfile mockRegistrationRequestProfile;
+
+    @Mock
+    RegistrationResponseProfile mockRegistrationResponseProfile;
+
+    @Mock
+    DCRManagementService mockDCRManagementService;
+
+    @BeforeMethod
+    public void setUp() {
+        initMocks(this);
```

Comment:
you don't need to call this method since you have already extend **PowerMockIdentityBaseTest**